### DISCUSSION TITLE
Floating point `f0` and `d0` suffixes for exponential notation?

### DISCUSSION THREAD

**Discussion by ornamentist** [coalton-lang/coalton#593] [Help!] [answered]
We can fix the type of a floating point literal with an `f0` or `d0` suffix:

```
(define float-x 123.456f0)

(define double-x 123.456d0)
```

But this isn't supported for floating point constants in exponential form?

```
(define exp-float-x 123.4e-10f0) ;; error

(define exp-double-x 123.4d10d0) ;; error
```

**Comment by eliaslfox** [comment, marked as answer]
Surprisingly, in common lisp `f` and `d` are exponent markers. 

```
(define exp-float-x 12.4f-10)

(define exp-double-x 123.4d10)
```

**Reply by ornamentist** [reply]
Aha -- that explains it.

**Reply by stylewarning** [reply]
And note that the exponent marker `e`/`E` is interpreted depending on the variable `cl:*read-default-float-format*`, so it's best not to use it. Stick to `f` and `d`.

**Reply by ornamentist** [reply]
Will do.
